nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -131,6 +136,12 @@ StaticLayer::getParameters()
   node_->get_parameter("unknown_cost_value", unknown_cost_value_);
   node_->get_parameter("trinary_costmap", trinary_costmap_);
 
+  if (!first_map_from_service_ && first_map_only_) {
```

Comment:
Can we reduce this to one parameter? Seems we could either get one map through the service interface or get continual updates through the topic. Is there a use case for getting an initial map through the service interface but want to get updates off the topic?

Reply:
I don't know why we even a map service in the first place for what can be done better with a simple topic. Ideally, I'd just switch it over, but I didn't want to remove functionality that was present before. I did it this way so when nav2 removes the map service, you just have to remove the first_map_from_service parameter.

Thoughts?

Reply:
I'm ok with getting rid of the map service usage here if we can ensure that nothing breaks if we get into operational mode without map data. I can't think of a situation where we need to get data through the service call anymore, since map server and cartographer provide the map on the topic.

@mjeronimo @bpwilcox What do you two think? I don't remember how we ended up with the service interface. Did that come from ROS 1 or did we add it?

Reply:
There was a service for the map as well as a latched topic for the map in the ROS 1 navigation.

Reply:
In general, given the trouble we've had with services, I'm OK with reducing the reliance on services, provided we don't introduce issues for determinism.